src/AzureAuth/Program.cs

```diff
@@ -35,14 +36,26 @@ private static void Main(string[] args)
             TelemetryOutput backend = TelemetryOutput.StandardOut;
 
             // We will only send telemetry if we are given an ingestion token via
-            // environment variable. We *have* to do this here, rather than in
+            // environment variable or registry. We *have* to do this here, rather than in
             // `CommandMain.OnExecute`, because Lasso doesn't have a way of allowing a
             // `CommandLineApplication` to dynamically set telemetry configuration. Even if
             // it did, we can't guarantee that a failure in command line parsing wouldn't
             // trigger telemetry before we ever get to disable it.
             //
             // To disable telemetry a user need only leave this environment variable unset. It's off by default.
             string applicationInsightsIngestionToken = Environment.GetEnvironmentVariable(EnvVars.ApplicationInsightsIngestionTokenEnvVar);
+
+            if (string.IsNullOrEmpty(applicationInsightsIngestionToken))
+            {
+                if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
+                {
+                    applicationInsightsIngestionToken = Win32.Registry.GetValue(
+                        EnvVars.ApplicationInsightsIngestionTokenRegKeyPath,
+                        EnvVars.ApplicationInsightsIngestionTokenRegKeyName,
+                        null) as string;

```

Comment:
What is the behavior of this when the value is unset?

Reply:
Is it worth having a try-catch here? Can this throw? I ask because this is outside the normal Lasso error catching boundary, and we shouldn't crash if we fail to get this. We can log errors to stderr, but we shouldn't crash.

Reply:
With the potential advent of error handling around this, I think it would also be worth pulling this into it's own static method to keep main as small and easy to read as possible.

Reply:
We don't need to add try-catch. If the key doesn't exist or is empty, `Registry.GetValue` will return null or empty string.